agent/dockerclient/dockerapi/docker_client_test.go

```diff
@@ -75,13 +78,17 @@ func dockerClientSetupWithConfig(t *testing.T, conf config.Config) (
 	func()) {
 	ctrl := gomock.NewController(t)
 	mockDocker := mock_dockeriface.NewMockClient(ctrl)
+	mockDockerSDK := mock_sdkclient.NewMockClient(ctrl)
```

Comment:
Its kind of weird alternating context on each line. I'd rather do one of these:

1) duplicate this file and have two sets of the same tests (so that we only ever test one thing per function)

2) use [subtests](https://blog.golang.org/subtests) and group the code into blocks instead of alternating each line?

```
func TestDockerFeature(t *testing.T) { {
    t.Run("sdk", func(t *testing.T) {
        // do sdk things
    })
    t.Run("docker", func(t *testing.T) {
        // old api things...
    })
}

Reply:
Idea behind this was that for most of these functions, a mock of NewDockerGoClient is created which requires both a clientfactory and an sdkClientFactory. Subtests looks like a great idea.